# Link to the earlier steps: In the earlier steps, we prepared the data and analyzed differnet calssification and regression models. As the optimal models are selected in the earlier steps, we will perform the steps as below here:
1. Based on the final files exported, we will reperform the finalist classification and regression models
2. Store the final models information using Pickle
3. Load the stored model and test it that it gives the same result as the originally trained model
4. Perform Prediction based on data in a csv file
Please refer to the "Data Preparation, Analysis and Artificial Intelligence or Machine Learning Analysis" file for the earlier steps referred to above.

In [18]:
# Import the required libraries and dependencies
import os
import requests
import json
import pandas as pd
from dotenv import load_dotenv
import matplotlib.pyplot as plt
import csv
from pathlib import Path
import numpy as np
import time
%matplotlib inline

In [19]:
#Import tha required libraries for datapreparation, analysis and preprocessing
import sklearn
import numpy as np
import seaborn as sns
from sklearn.preprocessing import OneHotEncoder
from sklearn.compose import make_column_transformer

In [20]:
#Import Required modules for Linear Regression
from sklearn.model_selection import train_test_split

# Import Metrics
from sklearn.metrics import accuracy_score
from sklearn.metrics import precision_score
from sklearn.metrics import recall_score

#Import different Models -  1. linera_model - LogisticRegression
#Import different Models -  5. svm - LinearSVC

from sklearn.linear_model import LogisticRegression
from sklearn.svm import LinearSVC 
#from sklearn.preprocessing import GetDummies
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from sklearn.discriminant_analysis import QuadraticDiscriminantAnalysis
#Linear Regression Library
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score
from sklearn.ensemble import RandomForestRegressor

In [21]:
#Import tha relevant alpha vantage libraries
from alpha_vantage.timeseries import TimeSeries
from alpha_vantage.cryptocurrencies import CryptoCurrencies

In [22]:
#set key
key = '32UEEELX55R5WBXX'
start_date = '2015-04-01' 

# Classification

## Read Data from preprocessed file, Prepare Data and Run Classification ML with the selected AI/ML Model -  Random Forest Classifier

In [23]:
#Load the stock analysis data for classification analysis
csv_path = Path("Resources/stock_analysis_data.csv")
stock_analysis_data = pd.read_csv(csv_path, index_col = 'date')
stock_analysis_data

value   EPS  treausry_yield  inflation_expectation  sp500_return  \
date                                                                           
2015-06   28.5798  3.84        0.090909              -0.035714      0.022148   
2015-06  540.0400  3.84        0.090909              -0.035714      0.022148   
2015-06  520.5100  3.84        0.090909              -0.035714      0.022148   
2015-06   39.3470  3.84        0.090909              -0.035714      0.022148   
2015-06  434.0900  3.84        0.090909              -0.035714      0.022148   
...           ...   ...             ...                    ...           ...   
2022-02   59.6512  3.35        0.175325               0.000000      0.037556   
2022-02  136.2333  3.35        0.175325               0.000000      0.037556   
2022-02  102.7000  3.35        0.175325               0.000000      0.037556   
2022-02   68.1660  3.35        0.175325               0.000000      0.037556   
2022-02  120.1474  3.35        0.175325               0.000000      0.037556   

         unemployment_rate  consumer_sentiment  %-close_AAPL-close  \
date                                                                 
2015-06          -0.053571            0.059537                 1.0   
2015-06          -0.053571            0.059537                 0.0   
2015-06          -0.053571            0.059537                 0.0   
2015-06          -0.053571            0.059537                 0.0   
2015-06          -0.053571            0.059537                 0.0   
...                    ...                 ...                 ...   
2022-02          -0.050000           -0.065476                 0.0   
2022-02          -0.050000           -0.065476                 0.0   
2022-02          -0.050000           -0.065476                 0.0   
2022-02          -0.050000           -0.065476                 0.0   
2022-02          -0.050000           -0.065476                 0.0   

         %-close_ABBV-close  %-close_ABT-close  ...  %-close_PM-close  \
date                                            ...                     
2015-06                 0.0                0.0  ...               0.0   
2015-06                 0.0                0.0  ...               0.0   
2015-06                 0.0                0.0  ...               0.0   
2015-06                 0.0                0.0  ...               0.0   
2015-06                 0.0                0.0  ...               0.0   
...                     ...                ...  ...               ...   
2022-02                 0.0                0.0  ...               0.0   
2022-02                 0.0                0.0  ...               0.0   
2022-02                 0.0                0.0  ...               0.0   
2022-02                 0.0                0.0  ...               0.0   
2022-02                 0.0                1.0  ...               0.0   

         %-close_T-close  %-close_UNH-close  %-close_UTX-close  \
date                                                             
2015-06              0.0                0.0                0.0   
2015-06              0.0                0.0                0.0   
2015-06              0.0                0.0                0.0   
2015-06              0.0                0.0                0.0   
2015-06              0.0                0.0                0.0   
...                  ...                ...                ...   
2022-02              0.0                0.0                0.0   
2022-02              0.0                0.0                0.0   
2022-02              0.0                0.0                1.0   
2022-02              0.0                0.0                0.0   
2022-02              0.0                0.0                0.0   

         %-close_V-close  %-close_VZ-close  %-close_WFC-close  \
date                                                            
2015-06              0.0               0.0                0.0   
2015-06              0.0               0.0              

In [24]:
# Listing of the features columns 
FEATURES = list(stock_analysis_data.iloc[:,:-1])
FEATURES

['value',
 'EPS',
 'treausry_yield',
 'inflation_expectation',
 'sp500_return',
 'unemployment_rate',
 'consumer_sentiment',
 '%-close_AAPL-close',
 '%-close_ABBV-close',
 '%-close_ABT-close',
 '%-close_AMGN-close',
 '%-close_AMZN-close',
 '%-close_BA-close',
 '%-close_BAC-close',
 '%-close_BMY-close',
 '%-close_BRK.B-close',
 '%-close_C-close',
 '%-close_CMCSA-close',
 '%-close_CSCO-close',
 '%-close_CVX-close',
 '%-close_DIS-close',
 '%-close_DWDP-close',
 '%-close_FB-close',
 '%-close_GE-close',
 '%-close_GILD-close',
 '%-close_GOOG-close',
 '%-close_GOOGL-close',
 '%-close_HD-close',
 '%-close_HON-close',
 '%-close_IBM-close',
 '%-close_INTC-close',
 '%-close_JNJ-close',
 '%-close_JPM-close',
 '%-close_KO-close',
 '%-close_MA-close',
 '%-close_MCD-close',
 '%-close_MDT-close',
 '%-close_MMM-close',
 '%-close_MO-close',
 '%-close_MRK-close',
 '%-close_MSFT-close',
 '%-close_NFLX-close',
 '%-close_NKE-close',
 '%-close_NVDA-close',
 '%-close_ORCL-close',
 '%-close_PEP-close',
 '%-clo

In [25]:
# Prepare test and train data
test_frac = 0.2
X = stock_analysis_data[FEATURES]
y = stock_analysis_data['Target']
x_train, x_test, y_train, y_test = train_test_split(X,y, test_size = test_frac)
x_train

value   EPS  treausry_yield  inflation_expectation  sp500_return  \
date                                                                           
2015-07   79.0932  3.84       -0.029762               0.037037     -0.061345   
2021-10  166.6035  2.52        0.290698               0.043478     -0.008035   
2020-01   98.6824  4.71       -0.071429               0.086957     -0.079166   
2017-05  162.6543  2.38        0.010989               0.040000      0.006374   
2021-07  119.6129  2.33       -0.095238               0.119048      0.029760   
...           ...   ...             ...                    ...           ...   
2019-02   57.9864  4.87       -0.019685              -0.037037      0.018079   
2018-11  100.1212  3.42       -0.016667              -0.034483     -0.087936   
2020-07  107.5600  2.18       -0.176471               0.000000      0.069797   
2016-06  144.7900  2.95       -0.100000               0.083333      0.036471   
2016-05   96.7997  2.35        0.031746              -0.142857      0.003497   

         unemployment_rate  consumer_sentiment  %-close_AAPL-close  \
date                                                                 
2015-07          -0.018868           -0.031217                 0.0   
2021-10          -0.021277           -0.015110                 0.0   
2020-01          -0.027778            0.005035                 0.0   
2017-05           0.000000            0.001031                 0.0   
2021-07          -0.084746           -0.050292                 0.0   
...                    ...                 ...                 ...   
2019-02          -0.050000            0.028509                 0.0   
2018-11           0.000000           -0.011156                 0.0   
2020-07          -0.072727           -0.071703                 0.0   
2016-06           0.020833           -0.012672                 0.0   
2016-05          -0.058824            0.064045                 0.0   

         %-close_ABBV-close  %-close_ABT-close  ...  %-close_PG-close  \
date                                            ...                     
2015-07                 0.0                0.0  ...               0.0   
2021-10                 0.0                0.0  ...               0.0   
2020-01                 0.0                0.0  ...               0.0   
2017-05                 0.0                0.0  ...               0.0   
2021-07                 0.0                1.0  ...               0.0   
...                     ...                ...  ...               ...   
2019-02                 0.0                0.0  ...               0.0   
2018-11                 0.0                0.0  ...               0.0   
2020-07                 0.0                0.0  ...               0.0   
2016-06                 0.0                0.0  ...               0.0   
2016-05                 0.0                0.0  ...               0.0   

         %-close_PM-close  %-close_T-close  %-close_UNH-close  \
date                                                            
2015-07               0.0              0.0                0.0   
2021-10               0.0              0.0                0.0   
2020-01               0.0              0.0                0.0   
2017-05               0.0              0.0                1.0   
2021-07               0.0              0.0                0.0   
...                   ...              ...                ...   
2019-02               0.0              0.0                0.0   
2018-11               0.0              0.0                0.0   
2020-07               0.0              0.0                0.0   
2016-06               0.0              0.0                0.0   
2016-05               0.0              0.0                0.0   

         %-close_UTX-close  %-close_V-close  %-close_VZ-close  \
date                                                            
2015-07                0.0              0.0               0.0   
2021-10                0.0              0.0               0.0   
2020

In [26]:

stock_logistic_regression_model = LogisticRegression(solver = 'liblinear')
stock_linear_regression_model.fit(x_train, y_train)
y_pred = stock_linear_regression_model.predict(x_test)

In [27]:

acc = accuracy_score (y_test, y_pred, normalize = True)
num_acc = accuracy_score (y_test, y_pred, normalize = False)
prec = precision_score (y_test, y_pred)
recall = recall_score (y_test, y_pred)

display(acc)
display(prec)
display(recall)

0.5538847117794486

0.5692068429237948

0.8224719101123595

In [30]:
# Define the RandomForestClassifier Model, Fit and Predict
stock_random_forest_classifirer_model = RandomForestClassifier()
stock_random_forest_classifirer_model.fit(x_train, y_train)
y_pred = stock_random_forest_classifirer_model.predict(x_test)

In [31]:
# Analyze the resuls of the RandomForestClassifier Model
rf_acc = accuracy_score (y_test, y_pred, normalize = True)
rf_num_acc = accuracy_score (y_test, y_pred, normalize = False)
rf_prec = precision_score (y_test, y_pred)
rf_recall = recall_score (y_test, y_pred)

display(rf_acc)
display(rf_prec)
display(rf_recall)

0.681704260651629

0.6758747697974218

0.8247191011235955

## Store the RamdomClassifier model and Reload the saved model

In [32]:
# Import Pickle 
import pickle

In [34]:
# Save the model
filename = 'random_forest_classifier.sav'
pickle.dump(stock_random_forest_classifirer_model, open(filename, 'wb'))

In [38]:
#Load the saved model
loaded_model = pickle.load(open('random_forest_classifier.sav','rb'))

In [49]:
#Predict using the loaded model and compare with the orginal model
y1_pred = loaded_model.predict(x_test)

NameError: name 'ytest' is not defined

In [50]:
y_test

date
2021-12    1.0
2016-06    1.0
2021-07    1.0
2017-06    0.0
2020-01    1.0
          ... 
2021-10    0.0
2018-01    1.0
2021-08    0.0
2020-05    0.0
2015-09    0.0
Name: Target, Length: 798, dtype: float64

In [40]:
#Predict using the loaded model and compare with the orginal model to confirm that the original model is the same as the loaded model.
rf_acc = accuracy_score (y_test, y_pred, normalize = True)
rf_num_acc = accuracy_score (y_test, y_pred, normalize = False)
rf_prec = precision_score (y_test, y_pred)
rf_recall = recall_score (y_test, y_pred)

display(rf_acc)
display(rf_prec)
display(rf_recall)

0.681704260651629

0.6758747697974218

0.8247191011235955

## Predict based on CSV file using the Loaded Model

In [42]:
# Load the file (Supposedly obtained from the Investor) from csv format
csv_path = Path("Resources/Classifier_Test.csv")
stock_test = pd.read_csv(csv_path, index_col = 'date', header=1)

In [43]:
#Review the analysis file
stock_test

value   EPS  treausry_yield  inflation_expectation  sp500_return  \
date                                                                           
2022-02  120.1474  3.35        0.175325                    0.0      0.037556   

         unemployment_rate  consumer_sentiment  %-close_AAPL-close  \
date                                                                 
2022-02              -0.05           -0.065476                 0.0   

         %-close_ABBV-close  %-close_ABT-close  ...  %-close_PG-close  \
date                                            ...                     
2022-02                 0.0                1.0  ...               0.0   

         %-close_PM-close  %-close_T-close  %-close_UNH-close  \
date                                                            
2022-02               0.0              0.0                0.0   

         %-close_UTX-close  %-close_V-close  %-close_VZ-close  \
date                                                            
2022-02                0.0              0.0               0.0   

         %-close_WFC-close  %-close_WMT-close  %-close_XOM-close  
date                                                              
2022-02                0.0                0.0                0.0  

[1 rows x 57 columns]

In [47]:
# Predict based on loaded model
prediction = loaded_model.predict(stock_test)

In [48]:
# Print
print(prediction)

[0.]


In [53]:
#Result of the analysis
if prediction [0] == 0:
    print('Do not buy the stocks')
else:
    print('Buy the stocks')

Buy the stocks


# Regression

## Read Data from preprocessed file, Prepare Data and Run Regression ML with the selected AI/ML Model -  Random Forest Regression

In [85]:
#Load the stock analysis data for classification analysis
csv_path = Path("Resources/stock_analysis_data_lr.csv")
stock_analysis_data_lr = pd.read_csv(csv_path, index_col = 'date')
stock_analysis_data

EPS  treausry_yield  inflation_expectation  sp500_return  \
date                                                                 
2015-06  3.84        0.090909              -0.035714      0.022148   
2015-06  3.84        0.090909              -0.035714      0.022148   
2015-06  3.84        0.090909              -0.035714      0.022148   
2015-06  3.84        0.090909              -0.035714      0.022148   
2015-06  3.84        0.090909              -0.035714      0.022148   
...       ...             ...                    ...           ...   
2022-02  3.35        0.175325               0.000000      0.037556   
2022-02  3.35        0.175325               0.000000      0.037556   
2022-02  3.35        0.175325               0.000000      0.037556   
2022-02  3.35        0.175325               0.000000      0.037556   
2022-02  3.35        0.175325               0.000000      0.037556   

         unemployment_rate  consumer_sentiment  %-close_AAPL-close  \
date                                                                 
2015-06          -0.053571            0.059537                 1.0   
2015-06          -0.053571            0.059537                 0.0   
2015-06          -0.053571            0.059537                 0.0   
2015-06          -0.053571            0.059537                 0.0   
2015-06          -0.053571            0.059537                 0.0   
...                    ...                 ...                 ...   
2022-02          -0.050000           -0.065476                 0.0   
2022-02          -0.050000           -0.065476                 0.0   
2022-02          -0.050000           -0.065476                 0.0   
2022-02          -0.050000           -0.065476                 0.0   
2022-02          -0.050000           -0.065476                 0.0   

         %-close_ABBV-close  %-close_ABT-close  %-close_AMGN-close  ...  \
date                                                                ...   
2015-06                 0.0                0.0                 0.0  ...   
2015-06                 0.0                0.0                 0.0  ...   
2015-06                 0.0                0.0                 0.0  ...   
2015-06                 0.0                0.0                 0.0  ...   
2015-06                 0.0                0.0                 0.0  ...   
...                     ...                ...                 ...  ...   
2022-02                 0.0                0.0                 0.0  ...   
2022-02                 0.0                0.0                 0.0  ...   
2022-02                 0.0                0.0                 0.0  ...   
2022-02                 0.0                0.0                 0.0  ...   
2022-02                 0.0                1.0                 0.0  ...   

         %-close_T-close  %-close_UNH-close  %-close_UTX-close  \
date                                                             
2015-06              0.0                0.0                0.0   
2015-06              0.0                0.0                0.0   
2015-06              0.0                0.0                0.0   
2015-06              0.0                0.0                0.0   
2015-06              0.0                0.0                0.0   
...                  ...                ...                ...   
2022-02              0.0                0.0                0.0   
2022-02              0.0                0.0                0.0   
2022-02              0.0                0.0                1.0   
2022-02              0.0                0.0                0.0   
2022-02              0.0                0.0                0.0   

         %-close_V-close  %-close_VZ-close  %-close_WFC-close  \
date                                                            
2015-06              0.0               0.0                0.0   
2015-06              0.0               0.0                0.0   
2015-06              0.0               0.0                0.0   
2015-06              0.0    

In [86]:
# Listing of the features columns 
FEATURES = list(stock_analysis_data_lr.iloc[:,:-1])
FEATURES

['EPS',
 'treausry_yield',
 'inflation_expectation',
 'sp500_return',
 'unemployment_rate',
 'consumer_sentiment',
 '%-close_AAPL-close',
 '%-close_ABBV-close',
 '%-close_ABT-close',
 '%-close_AMGN-close',
 '%-close_AMZN-close',
 '%-close_BA-close',
 '%-close_BAC-close',
 '%-close_BMY-close',
 '%-close_BRK.B-close',
 '%-close_C-close',
 '%-close_CMCSA-close',
 '%-close_CSCO-close',
 '%-close_CVX-close',
 '%-close_DIS-close',
 '%-close_DWDP-close',
 '%-close_FB-close',
 '%-close_GE-close',
 '%-close_GILD-close',
 '%-close_GOOG-close',
 '%-close_GOOGL-close',
 '%-close_HD-close',
 '%-close_HON-close',
 '%-close_IBM-close',
 '%-close_INTC-close',
 '%-close_JNJ-close',
 '%-close_JPM-close',
 '%-close_KO-close',
 '%-close_MA-close',
 '%-close_MCD-close',
 '%-close_MDT-close',
 '%-close_MMM-close',
 '%-close_MO-close',
 '%-close_MRK-close',
 '%-close_MSFT-close',
 '%-close_NFLX-close',
 '%-close_NKE-close',
 '%-close_NVDA-close',
 '%-close_ORCL-close',
 '%-close_PEP-close',
 '%-close_PFE-clo

In [104]:
# Prepare test and train data
test_frac = 0.2
Xlr = stock_analysis_data_lr[FEATURES]
ylr = stock_analysis_data_lr['value']
x_train_lr, x_test_lr, y_train_lr, y_test_lr = train_test_split(Xlr,ylr, test_size = test_frac)
x_train_lr

EPS  treausry_yield  inflation_expectation  sp500_return  \
date                                                                 
2020-10  2.58        0.259259               0.000000      0.108777   
2017-07  2.97        0.056497               0.000000      0.002918   
2021-06  2.33        0.024390              -0.086957      0.024413   
2017-10  3.30        0.100000              -0.111111      0.030566   
2019-08  3.17       -0.185792               0.038462      0.019479   
...       ...             ...                    ...           ...   
2018-07  3.08        0.000000              -0.033333      0.031920   
2020-03  1.84       -0.553030              -0.083333      0.126984   
2016-08  2.95        0.056075              -0.074074      0.000078   
2018-07  3.08        0.000000              -0.033333      0.031920   
2019-04  2.25       -0.016878               0.000000     -0.063771   

         unemployment_rate  consumer_sentiment  %-close_AAPL-close  \
date                                                                 
2020-10          -0.126582            0.017413                 0.0   
2017-07           0.000000           -0.016842                 0.0   
2021-06           0.017241            0.031363                 0.0   
2017-10          -0.023256            0.058885                 0.0   
2019-08           0.000000           -0.087398                 0.0   
...                    ...                 ...                 ...   
2018-07          -0.050000           -0.003055                 0.0   
2020-03           0.257143           -0.117822                 0.0   
2016-08           0.020833           -0.002222                 0.0   
2018-07          -0.050000           -0.003055                 0.0   
2019-04          -0.052632           -0.012195                 0.0   

         %-close_ABBV-close  %-close_ABT-close  %-close_AMGN-close  ...  \
date                                                                ...   
2020-10                 0.0                0.0                 0.0  ...   
2017-07                 0.0                0.0                 0.0  ...   
2021-06                 0.0                0.0                 0.0  ...   
2017-10                 0.0                0.0                 0.0  ...   
2019-08                 1.0                0.0                 0.0  ...   
...                     ...                ...                 ...  ...   
2018-07                 0.0                0.0                 1.0  ...   
2020-03                 0.0                0.0                 0.0  ...   
2016-08                 0.0                0.0                 0.0  ...   
2018-07                 0.0                0.0                 0.0  ...   
2019-04                 0.0                0.0                 0.0  ...   

         %-close_PM-close  %-close_T-close  %-close_UNH-close  \
date                                                            
2020-10               0.0              0.0                0.0   
2017-07               0.0              1.0                0.0   
2021-06               0.0              0.0                0.0   
2017-10               0.0              0.0                0.0   
2019-08               0.0              0.0                0.0   
...                   ...              ...                ...   
2018-07               0.0              0.0                0.0   
2020-03               0.0              0.0                0.0   
2016-08               0.0              0.0                0.0   
2018-07               0.0              0.0                0.0   
2019-04               0.0              0.0                0.0   

         %-close_UTX-close  %-close_V-close  %-close_VZ-close  \
date                                                            
2020-10                0.0              0.0               0.0   
2017-07                0.0              0.0               0.0   
2021-06                0.0              0.0               0.0   
2017-10                0.0              0

In [105]:
# Define the RandomForestRegression Model, Fit and Predict
stock_random_forest_regression_model = RandomForestRegressor()
stock_random_forest_regression_model.fit(x_train_lr, y_train_lr)
ylr_pred = stock_random_forest_regression_model.predict(x_test_lr)
ylr_pred

array([ 110.87592 ,   33.576845,  122.49665 ,   49.943078,   20.8362  ,
         49.457413,  187.636   ,  212.44745 , 1065.8201  ,   77.976477,
        172.333903,  234.883342,   48.939077,  100.922576,  124.17144 ,
         72.597424,   46.002542,   25.84145 ,  201.697937,   48.461933,
        183.697944,  211.386199,  135.112679,   69.752812,   52.440616,
        330.417663,   55.529199,   59.363737,   38.865786,   53.303822,
         46.929113,   61.536062,   65.994533,  110.350564,  207.781614,
         56.353299,   74.568315,  189.316   ,  160.4629  ,  155.530379,
         57.015875,   47.885161,   53.26802 ,   46.061719,   40.757892,
        978.1817  , 1449.9937  ,  179.988   ,  151.18665 , 1107.5483  ,
        263.065347,  153.168229,   19.434092,   33.778912,  206.622743,
        113.086672,   88.017712,   69.57528 ,  202.4502  ,  121.757918,
         85.161593,  110.107919,   93.161846,   87.100192,   95.814314,
        751.3246  ,   32.164928,   44.969019,  218.6968  ,   41.

In [108]:
r21_score = r2_score(y_test_lr, ylr_pred, multioutput='variance_weighted')
print(r21_score)

TypeError: 'numpy.float64' object is not callable

## Store the RamdomRegression model and Reload the saved model

In [109]:
#Save the regressor model
filename = 'random_forest_regressor.sav'
pickle.dump(stock_random_forest_regression_model, open(filename, 'wb'))

In [110]:
#loading the saved model
loaded_model_lr = pickle.load(open('random_forest_regressor.sav','rb'))

value
59.6512
136.2333
102.7
68.166
120.1474

## Predict based on CSV file using the Loaded Model

In [111]:
# Load the file (Supposedly obtained from the Investor) from csv format
csv_path = Path("Resources/stock_test_regression.csv")
stock_test_regr = pd.read_csv(csv_path, index_col = 'date', header=1)
stock_test_regr

EPS  treausry_yield  inflation_expectation  sp500_return  \
date                                                                 
2022-02  3.35        0.175325                    0.0      0.037556   
2022-02  3.35        0.175325                    0.0      0.037556   
2022-02  3.35        0.175325                    0.0      0.037556   
2022-02  3.35        0.175325                    0.0      0.037556   
2022-02  3.35        0.175325                    0.0      0.037556   

         unemployment_rate  consumer_sentiment  %-close_AAPL-close  \
date                                                                 
2022-02              -0.05           -0.065476                 0.0   
2022-02              -0.05           -0.065476                 0.0   
2022-02              -0.05           -0.065476                 0.0   
2022-02              -0.05           -0.065476                 0.0   
2022-02              -0.05           -0.065476                 0.0   

         %-close_ABBV-close  %-close_ABT-close  %-close_AMGN-close  ...  \
date                                                                ...   
2022-02                 0.0                0.0                 0.0  ...   
2022-02                 0.0                0.0                 0.0  ...   
2022-02                 0.0                0.0                 0.0  ...   
2022-02                 0.0                0.0                 0.0  ...   
2022-02                 0.0                1.0                 0.0  ...   

         %-close_PM-close  %-close_T-close  %-close_UNH-close  \
date                                                            
2022-02               0.0              0.0                0.0   
2022-02               0.0              0.0                0.0   
2022-02               0.0              0.0                0.0   
2022-02               0.0              0.0                0.0   
2022-02               0.0              0.0                0.0   

         %-close_UTX-close  %-close_V-close  %-close_VZ-close  \
date                                                            
2022-02                0.0              0.0               0.0   
2022-02                0.0              0.0               0.0   
2022-02                1.0              0.0               0.0   
2022-02                0.0              0.0               0.0   
2022-02                0.0              0.0               0.0   

         %-close_WFC-close  %-close_WMT-close  %-close_XOM-close  Target  
date                                                                      
2022-02                0.0                0.0                0.0     0.0  
2022-02                0.0                0.0                0.0     0.0  
2022-02                0.0                0.0                0.0     1.0  
2022-02                0.0                0.0                0.0     1.0  
2022-02                0.0                0.0                0.0     0.0  

[5 rows x 57 columns]

In [112]:
# Predict the stock values for the data obtained from the investor
regr_predictions = loaded_model_lr.predict(stock_test_regr)
regr_predictions

array([ 59.9283  , 126.619401,  64.600623,  64.587928, 112.40737 ])